src/xenia/kernel/kernel_state.cc

```diff
@@ -749,7 +750,7 @@ void KernelState::RegisterNotifyListener(XNotifyListener* listener) {
     uint32_t live_connection_state =
         cvars::offline_mode ? X_ONLINE_S_LOGON_DISCONNECTED
                             : X_ONLINE_S_LOGON_CONNECTION_ESTABLISHED;
-    uint32_t ethernet_link_state = cvars::offline_mode ? 0 : 1;
+    uint32_t ethernet_link_state = (cvars::offline_mode && !cvars::systemlink_always_allowed) ? 0 : 1;
```

Comment:
If offline_mode is true and systemlink_always_allowed is false, ethernet is down